### PR TITLE
Fixing a few a11y issues in the app

### LINKED ISSUE #377: Narrator is silent and focus lands to "Aqua" on pressing "Home" Key in the colors list in ItemsRepeater.
**Describe the bug**

Home and End keys are not handled in the repeater sample. 

**Steps to reproduce the bug**

1. Turn on Narrator.
2. Launch XAML Controls Gallery.
3. Activate ItemsRepeater from home page, navigate to header "Animated Scrolling and Content Display" and press  "Home" Key and observe.

**Expected behavior**

1. Focus should go to the first item in the list i.e "AliceBlue" 
2. Narrator should describe about the item.

**Actual behavior**

Focus does not land on the first list on pressing Home key but it's focus goes to  "Aqua" . Also, Narrator is silent.

**Screenshots**

image

**Version Info**

Gallery version (found on Settings page): 

1.2.13.0

**Additional context**

Copied from [internal bug](https://microsoft.visualstudio.com/OS/_workitems/edit/25469213)

### LINKED ISSUE #378: Animated Scrolling and Content Display selected button value gets changed when we re-enter again.
**Describe the bug**

**Steps to reproduce the bug**

1. Launch XAML Controls Gallery.
2. Activate Items Repeater from All Controls Page.
3. With tab navigation navigate to any button present under header "Animated Scrolling and Content Display", now select any color of your choice and then press tab.
4. Now press shift-tab again and check if the selected button is same as that was selected earlier.

**Expected behavior**

Selected button should not change.

**Actual behavior**

Selected button gets changed.

**Screenshots**

image

**Version Info**

Gallery version (found on Settings page): 

1.2.13.0

**Additional context**

Copied from [internal bug](https://microsoft.visualstudio.com/OS/_workitems/edit/25468901)

### PR DESCRIPTION
**Pull request by ranjeshj**
## Description
Fixing a handful of accessibility issues in the app.

## Motivation and Context
Fixes the following issues
Internal: [Bug 37700602](https://microsoft.visualstudio.com/DefaultCollection/OS/_workitems/edit/37700602): [XCG]ItemsRepeater does not meet contrast requirements in Aquatic High Contrast theme

Images after the fix in Normal and Aquatic themes.
image
image

Internal [Bug 38293529](https://microsoft.visualstudio.com/DefaultCollection/OS/_workitems/edit/38293529): [XCG] WinUI Accessibility->Connected Animation card: When any item is invoked, KB focus does not fall on the immediate interactive control(Go back button)
Setting focus on the correct item after the connected animation.

Internal [Bug 37020621](https://microsoft.visualstudio.com/DefaultCollection/OS/_workitems/edit/37020621): [XCG] WinUI 3 Controls Gallery->Connected Animation: Narrator does not announce Control type of different menu items
Setting missing accessibility name for the item.

Internal [Bug 36430620](https://microsoft.visualstudio.com/DefaultCollection/OS/_workitems/edit/36430620): [XCG][GitHub] Animated Scrolling and Content Display selected button value gets changed when we re-enter again. · Issue #378 · microsoft/Xaml-Controls-Gallery (github.com)
Fixes #378
We need to keep track of the last item that was focused so we put focus back on the same item.

Internal [Bug 36430694](https://microsoft.visualstudio.com/DefaultCollection/OS/_workitems/edit/36430694): [XCG][GitHub] Narrator is silent and focus lands to "Aqua" on pressing "Home" Key in the colors list in ItemsRepeater. #377
Fixes #377
we were not handling home and end keys 

## How Has This Been Tested?
Tested manually, visually as well as with Narrator/Accessibility Insights.

## Types of changes

- [ x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: XamlControlsGallery/ControlPages/ItemsRepeaterPage.xaml.cs [hunk 1]
```diff
@@ -402,7 +421,32 @@ private void UpdateSortAndFilter()
...
             peer.RaiseNotificationEvent(AutomationNotificationKind.Other, AutomationNotificationProcessing.ImportantMostRecent, $"Filtered recipes, {sortedFilteredTypes.Count()} results.", "RecipesFilteredNotificationActivityId");
-        }
+        }
+
+        private void OnAnimatedScrollRepeaterKeyDown(object sender, Windows.UI.Xaml.Input.KeyRoutedEventArgs e)
+        {
+            if (e.Handled != true)
+            {
+                var targetIndex = -1;
+                if (e.Key == Windows.System.VirtualKey.Home)
```
Comment (llongley) [inline XamlControlsGallery/ControlPages/ItemsRepeaterPage.xaml.cs:431]: Are these the only keys we need to handle here?  What about PgUp/PgDown, for example?

File: XamlControlsGallery/ControlPages/ItemsRepeaterPage.xaml [hunk 2]
```diff
@@ -31,15 +31,15 @@
         </DataTemplate>
 
         <DataTemplate x:Key="NormalItemTemplate" x:DataType="x:Int32">
-            <Button HorizontalAlignment="Stretch" VerticalAlignment="Stretch" Background="{ThemeResource SystemChromeLowColor}">
-                <TextBlock Text="{x:Bind}"/>
-            </Button>
+            <Border Background="{ThemeResource SystemControlBackgroundChromeMediumBrush}">
+                <TextBlock Text="{x:Bind}" HorizontalAlignment="Center" VerticalAlignment="Center" />
```
Comment (llongley) [inline XamlControlsGallery/ControlPages/ItemsRepeaterPage.xaml:35]: Why change Horizontal/VerticalAlignment here?

File: XamlControlsGallery/ConnectedAnimationPages/CardPage.xaml [hunk 3]
```diff
@@ -17,16 +17,16 @@
...
                             <RowDefinition Height="Auto" />
                             <RowDefinition Height="*" />
                         </Grid.RowDefinitions>
                         <Grid HorizontalAlignment="Stretch" VerticalAlignment="Stretch" Background="{ThemeResource SystemAccentColor}" Height="100">
-                            <TextBlock Text="Header" HorizontalAlignment="Center" VerticalAlignment="Center" Style="{ThemeResource TitleTextBlockStyle}" />
+                            <TextBlock Text="Item" HorizontalAlignment="Center" VerticalAlignment="Center" Style="{ThemeResource TitleTextBlockStyle}" />
                         </Grid>
                         <Grid HorizontalAlignment="Stretch" VerticalAlignment="Stretch" Background="{ThemeResource SystemControlBackgroundChromeMediumLowBrush}" Grid.Row="1">
-                            <TextBlock Text="Content" Margin="12" />
+                            <TextBlock Text="{Binding}" Margin="12" HorizontalAlignment="Center" VerticalAlignment="Center"/>
```
Comment (llongley) [inline XamlControlsGallery/ConnectedAnimationPages/CardPage.xaml:29]: Similar question about Horizontal/VerticalAlignment here too.

**Review by llongley — CHANGES_REQUESTED** [review]

File: XamlControlsGallery/ControlPages/ItemsRepeaterPage.xaml [hunk 4]
```diff
@@ -31,15 +31,15 @@
...
+            </Border>
         </DataTemplate>
 
         <DataTemplate x:Key="AccentItemTemplate" x:DataType="x:Int32">
-            <Button HorizontalAlignment="Stretch" VerticalAlignment="Stretch" Background="{ThemeResource SystemAccentColor}">
-                <TextBlock Text="{x:Bind}"/>
-            </Button>
+            <Border  Background="{ThemeResource SystemControlBackgroundAccentBrush}" >
+                <TextBlock Text="{x:Bind}"  HorizontalAlignment="Center" VerticalAlignment="Center" Foreground="{ThemeResource SystemControlForegroundChromeWhiteBrush}"/>
```
Comment (karenbtlai) [inline XamlControlsGallery/ControlPages/ItemsRepeaterPage.xaml:41]: Does the chrome white brush here work with both light/dark theme?

Comment (karenbtlai) [reply XamlControlsGallery/ControlPages/ItemsRepeaterPage.xaml:41]: I realized the brushes used in XCG are all pointing towards older brushes. I'll make a note go through them and update them to the new brushes recommended by design in the visual updates.

File: XamlControlsGallery/ControlPages/ItemsRepeaterPage.xaml.cs (hunk 1 above)
Comment (ranjeshj) [reply XamlControlsGallery/ControlPages/ItemsRepeaterPage.xaml.cs:431]: PageUp and PageDown are handled by the wrapping ScrollViewer. This is not a full fledged implementation of a collection scenario for sure.

File: XamlControlsGallery/ControlPages/ItemsRepeaterPage.xaml (hunk 2 above)
Comment (ranjeshj) [reply XamlControlsGallery/ControlPages/ItemsRepeaterPage.xaml:35]: I moved from Button to Border so the alignment changed. I added these to center the text back again.

File: XamlControlsGallery/ConnectedAnimationPages/CardPage.xaml (hunk 3 above)
Comment (ranjeshj) [reply XamlControlsGallery/ConnectedAnimationPages/CardPage.xaml:29]: This looks better than top/left aligned.

File: XamlControlsGallery/ControlPages/ItemsRepeaterPage.xaml (hunk 4 above)
Comment (ranjeshj) [reply XamlControlsGallery/ControlPages/ItemsRepeaterPage.xaml:41]: Here is what it looks like in dark theme. A border might help in high contrast theme.
image

**Pull request merged**